king_phisher/client/application.py

```diff
@@ -552,6 +555,19 @@ def do_config_load(self, load_defaults):
 			for key, value in client_template.items():
 				if not key in self.config:
 					self.config[key] = value
+		env_proxy = os.environ.get('HTTPS_PROXY')
+		if env_proxy is not None:
+			proxy_url = urllib.parse.urlparse(env_proxy)
+			if not (proxy_url.hostname and proxy_url.scheme):
+				self.logger.error('invalid proxy url (missing scheme or hostname)')
+				return
+			if env_proxy != self.config['proxy.url']:
+				self.logger.warning('proxy environment variable does not match the user configuration')
+			else:
+				self.logger.info('proxy environment variable is the same as the user configuration')
+			self.config['proxy.url'] = env_proxy
+		os.environ['HTTPS_PROXY'] = self.config['proxy.url']
```

Comment:
This is crashing on me when I don't have an environment variable already set. So if `self.config['proxy.url']` is `None`, this will crash because you can't set environment variables to `None`.

You might need another conditional statement to set the `HTTP(S)_PROXY` environment variables when `self.config['proxy.url']` is set. That should come after the check you're already done. If neither are set, then I would highly recommend deleting the `HTTP_PROXY` environment variable so we know that the configuration is synced and that particular value won't get leaked in.

This is what I mocked up until I got to the second error.

```
		if env_proxy is not None:
			#... do what you're doing already but here
		elif self.config['proxy.url']:
			os.environ['HTTPS_PROXY'] = self.config['proxy.url']
			os.environ['HTTP_PROXY'] = self.config['proxy.url']
		else:
                        # delete HTTP_PROXY if there is not a known configuration
			os.environ.pop('HTTP_PROXY', None)
```